.github/workflows/miniconda_templated.yaml

```diff
@@ -2,69 +2,82 @@ name: Build and Push miniconda templated containers
 
 on:
   push:
+    branches:
+      - master
```

Comment:
Shouldn't push only respond to "*/image_tag", or do you want to allow version over-writes?

Reply:
I was hoping to introduce an active test condition on that, so that the test fails and you cannot merge.

Reply:
I mostly mean to make things simpler. We don't need to push a new image unless the version has changed (unless allowing overlaps), so no other files matter.

Reply:
I can see situations where you might want to introduce multiple orthogonal changes without a new container each time, so those could not trigger a new container on quay.io, but we still check that they won't break anything.

Reply:
But would you want the tests to build the image even if there is no tag change?

Reply:
I have added now a test that, for every umamba folder, will enforce that the image_tag file is changed if that folder was detected to change.

Reply:
In other words, if you want that container rebuilt (and pushed) you need to change the tag.

Reply:
I was thinking that I might possibly want to check the image still builds for some PRs without triggering a new container 'release', e.g. where there were sets of unrelated changes to be introduced over multiple PRs. So we'd allow container builds (but not uploads) without a tag change to check they worked, and then when we were happy with the current state of things there would be a tag bump and an upload.

But this will work too- we'll just have more images on Quay!

Reply:
...mmm.. I don't see why you would want to check that it builds but not actually push it. If that is the objective you could open a PR but not merge it. Given its immutability, I'm guessing the only reason to try a re-build is because you actually need the container to be different, correct?

Reply:
Yes, you're right, it's getting done during the PR so we're good there.

For the other part I was just imagining scenarios where e.g. feature A and B are quite different, so part of separate PRs, but we don't want a container to exist with only Feature A, so we'd only bump the tag once we've done both A and B. 

But maybe an edge case.